src/vacuum-card.ts

```diff
@@ -189,23 +241,50 @@ export class VacuumCard extends LitElement {
       return nothing;
     }
 
-    const selected = sources.indexOf(source);
+    return this.renderDropDown(source, sources, 'mdi:fan', this.handleSpeed);
+  }
+
+  private renderWaterLevel(): Template {
+    const entity = this.waterLevel;
+
+    if (!entity) {
+      return nothing;
+    }
+
+    return this.renderDropDown(
+      entity.state,
+      entity.attributes.options,
+      'mdi:water',
+      this.handleSelect,
+    );
+  }
+
+  private renderDropDown(
+    selectedObject: string,
+    objects: {
+      [key: string]: any;
+    },
```

Comment:
Let's use `Record`:

```suggestion
    objects: Record<string, any>,
```

`object` is a reserved name in TS meaning a type of `Object`. Let's use `entries` or `iterms` or something.

```suggestion
    items: Record<string, any>,
```